### ISSUE TITLE
S3 select behavior different from AWS S3 behavior

### ISSUE THREAD

**Issue by harshavardhana** [minio/minio#6504]
*** not using template ***

From @badscooter23 

Found a bug where Minio - Select and S3 - Select return different answers on the same queries.

If I do the following query:

```
select count(*) from S3Object s where s.WeatherDelay <> '0' and s.WeatherDelay <> ''"
```

against S3 and Minio, I get the same answer (modulo the fact that it returns the answer in a different format which is a bug we have already filed)

```
select count(*) from S3Object s where s.WeatherDelay <> '0' and s.WeatherDelay <> ''
```

Host 's3' (https://s3.amazonaws.com)
1247488

Stats details bytesScanned:  247963212
Stats details bytesProcessed:  247963212
StartTime:  2018-09-20 09:40:02.299235
EndTime:  2018-09-20 09:40:06.794729
ElapsedTime:  0:00:04.495494

Host 'play' (https://play.minio.io:9000)
1247488.000000

Stats details bytesScanned:  247963212
Stats details bytesProcessed:  247963212
StartTime:  2018-09-20 09:40:06.795705
EndTime:  2018-09-20 09:41:42.551258
ElapsedTime:  0:01:35.755553

Host 'z0' (http://atl20ds9minio01.amer.prgx.com:9000)
1247488.000000

Stats details bytesScanned:  247963212
Stats details bytesProcessed:  247963212
StartTime:  2018-09-20 09:41:42.555645
EndTime:  2018-09-20 09:42:01.280830
ElapsedTime:  0:00:18.725185

Host 'm0' (http://atl20ds5minas01.amer.prgx.com:9000)
1247488.000000

Stats details bytesScanned:  247963212
Stats details bytesProcessed:  247963212
StartTime:  2018-09-20 09:42:01.282840
EndTime:  2018-09-20 09:42:23.909868
ElapsedTime:  0:00:22.627028 (edited)
as you can see all four systems (s3, play, m0 and z0 - where m0 and z0 are internal minio clusters) all return

1247488 records matching the query, minio returns the number as a "real" number (1247488.00000) but the numbers match You can also see from that query that S3 is MUCH faster than any of those Minio clusters...However, if I change the query just a little bit, then S3 and Minio return different answers

```
select count(*) from S3Object s where s.WeatherDelay <> '0.0' and s.WeatherDelay <> ''
```

Host 's3' (https://s3.amazonaws.com)
98918

Stats details bytesScanned:  247963212
Stats details bytesProcessed:  247963212
StartTime:  2018-09-20 09:47:08.593349
EndTime:  2018-09-20 09:47:11.768155
ElapsedTime:  0:00:03.174806

Host 'play' (https://play.minio.io:9000)
1247488.000000

Stats details bytesScanned:  247963212
Stats details bytesProcessed:  247963212
StartTime:  2018-09-20 09:47:11.772697
EndTime:  2018-09-20 09:48:47.510464
ElapsedTime:  0:01:35.737767

Host 'z0' (http://atl20ds9minio01.amer.prgx.com:9000)
1247488.000000

Stats details bytesScanned:  247963212
Stats details bytesProcessed:  247963212
StartTime:  2018-09-20 09:48:47.515151
EndTime:  2018-09-20 09:49:09.116059
ElapsedTime:  0:00:21.600908

Host 'm0' (http://atl20ds5minas01.amer.prgx.com:9000)
1247488.000000

Stats details bytesScanned:  247963212
Stats details bytesProcessed:  247963212
StartTime:  2018-09-20 09:49:09.118646
EndTime:  2018-09-20 09:49:32.098705
ElapsedTime:  0:00:22.980059

All DONE! (edited)
Here are the two queries side by side:

```
select count(*) from S3Object s where s.WeatherDelay <> '0' and s.WeatherDelay <> ''"
select count(*) from S3Object s where s.WeatherDelay <> '0.0' and s.WeatherDelay <> ''"
```

Minio returns the same thing for both queries - 1247488.000000.
S3 returns 1247488 for the first query and 98918 for the second one.

In all cases, Minio is much slower (with play and our internal minio clusters) than S3.

**Comment by harshavardhana** [comment]
```
mc ls m2/sjm-airlines/
[2018-09-20 12:04:35 PDT] 236MiB DelayedFlights.csv
```

CSV used is available here

**Comment by harshavardhana** [comment]
This issue is fixed in master we interpret now 0.0 as float and 0 as int  just like AWS S3
